lib/src/widgets/input/input_types/input_date_picker/date_picker.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:intl/intl.dart';
+import '../../../../../flutter_form.dart';
+
+/// Generates a [TextFormField] for passwords. It requires a [FlutterFormInputController]
+/// as the [controller] parameter and an optional [Widget] as [label]
+class DateTimeInputField extends ConsumerStatefulWidget {
+  const DateTimeInputField(
+      {Key? key,
+      required this.inputType,
+      required this.controller,
+      this.label,
+      this.showIcon = true,
+      this.icon = Icons.calendar_today,
+      required this.dateFormat})
+      : super(
+          key: key,
+        );
+  final FlutterFormDateTimeType inputType;
+  final FlutterFormInputController controller;
+  final DateFormat dateFormat;
+  final bool showIcon;
+  final IconData icon;
+  final Widget? label;
+  @override
+  ConsumerState<DateTimeInputField> createState() => _DateInputFieldState();
+}
+
+class _DateInputFieldState extends ConsumerState<DateTimeInputField> {
+  @override
+  Widget build(BuildContext context) {
+    String Function(String, {List<String>? params}) _ =
+        getTranslator(context, ref);
+
+    Future<String> getInputFromUser(FlutterFormDateTimeType inputType) async {
+      String userInput = '';
+      switch (inputType) {
+        case FlutterFormDateTimeType.date:
+          DateTime? unformatted = await showDatePicker(
+            context: context,
+            initialDate: DateTime.now(),
+            firstDate: DateTime.now(),
+            lastDate: DateTime.now().add(
+              const Duration(days: 10000),
+            ),
+          );
+          userInput = unformatted != null
+              ? widget.dateFormat.format(unformatted)
+              : userInput;
+          break;
+        case FlutterFormDateTimeType.dateTime:
+          await getInputFromUser(FlutterFormDateTimeType.date)
+              .then((value) async {
+            if (value != '') {
+              String secondInput =
+                  await getInputFromUser(FlutterFormDateTimeType.time);
+              if (secondInput != '') {
+                userInput = '$value $secondInput';
+              }
+            }
+          });
+          break;
+        case FlutterFormDateTimeType.range:
+          userInput = (await showDateRangePicker(
+            context: context,
+            firstDate: DateTime.now(),
+            lastDate: DateTime.now().add(
+              const Duration(days: 10000),
+            ),
+          ).then((value) {
+            return value != null
+                ? '${widget.dateFormat.format(value.start)} - ${widget.dateFormat.format(value.end)}'
+                : '';
+          }))
+              .toString();
+          break;
+        case FlutterFormDateTimeType.time:
+          userInput = await showTimePicker(
+                  context: context, initialTime: TimeOfDay.now())
+              .then((value) => value == null ? '' : value.format(context));
+      }
+      return userInput;
+    }
+
+    return TextFormField(
```

Comment:
Add the following parameter: keyboardType: TextInputType.none,

This makes it so the keyboard doesn't open at all.